### DISCUSSION TITLE
Resetting the datagrid pager

### DISCUSSION THREAD

**Discussion by GilShalit** [Megabit/Blazorise#2323] [Q&A]
I am using the DataGrid to display and edit a large dataset, so I am calling my API in the OnReadData event, with e.Page and e.PageSize.  I am also capturing changes to the sorting columns by inspecting e.Columns  and building parameters for the API to get the data sorted accordingly.  

If the sorting columns or their directions change I think it makes sense to go to the first page with the new sorting parameters, and I can easily call my API with page=1 instead of e.Page.  But how do I tell the pager we are on the first page?

Thanks in advance!

**Comment by stsrki** [comment]
You only need to set `CurrentPage` parameter. It should be enough to reset the pager.

**Reply by GilShalit** [reply]
Thanks.  The problem with that is the if CurrentPage is changed in code it results in re-entry of OnReadData(), so I have to make sure the API is only called once - only on the re-entry and not the initial run-through of the event.

Easily done, thanks again!
